### ISSUE TITLE
Popcorn Palace: Obfuscates object names

### ISSUE THREAD

**Issue by regalialong** [AdGoBye/AdGoBye-Blocklists#11]
First report per Blocklist Callback on 2024-08-14, they have started scrambling Gameobject names, which means we'll not be able to get good blocks for this.

We'll require a more holistic approach for this, I'll play the cat and mouse game for now.

**Comment by regalialong** [comment]
Nope, as expected, the names rotate. This'll require a holistic solution.

**Comment by SRThrowaway** [comment]
Either they forgot to obscure the latest update, or they get mostly unobscured at runtime. Enabled ads, excluding ads using canvases or sprites, had normal names while disabled ones were still random. Might be worth looking into a script to run the world and grab the unobscured names.

**Comment by regalialong** [comment]
The names on my end are still obfuscated, same with the logs so I don't think they're cleaned up at runtime either. I have a PoC to deal with this but I have no ETA on it, I'm pretty busy right now.

**Comment by SRThrowaway** [comment]
![image](https://github.com/user-attachments/assets/66d57c08-734f-4bb6-b21b-f0501d72d583)
This is what I'm seeing. I think it dynamically enables/disables based on what the server returns.

**Comment by regalialong** [comment]
For what it's worth, you can probably check the log and see if it pulls the Posters from a separate domain. AGB doesn't do domain / network blocking but if it image loads the posters, that should fix that.

I think at most from AGB / Blocklists side, you could disable `Z9UQEMT` since I assume that probably contains the actual loader script. 

The problem is that the `Z9UQEMT`s name will probably rotate on next update, Blocklists assumes object names stay static between versions. Updating names for every update is treadmill work I'm not very interested in, so we need a new approach to detect those objects, which I already got the code for, it mostly requires plumbing work.
